### PR TITLE
Refactor Resume, Submission, and Dashboard

### PR DESCRIPTION
**Pull request by toyoun**
There are temporary changes in this to make the application submission function properly. I will update / undo them once the to-do items below are completed.

TODO:
1. Update the backend to initialize the application object with a `status` variable with the initial value of `0` to display application status as `Undecided`
2. Update data types in Application Model to more legible types:
- `gender` from `Number` to `String`
- `grad_date` from one variable to two (i.e. `grad_month` and `grad_year` of types `String` and `Number` respectively)
- `ethnicity` from `Number` to `String`
- `diet` from `Number` to `String`

### CODE REVIEW

File: src/components/InputTypes/FileInput.jsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ class FileInput extends Component {
             {label}
           </FormLabel>
         </div>
-        <input style={{ display: 'none' }} id="files" accept=".pdf" type="file" onChange={this.onChange} onBlur={() => {}} disabled={disabled} />
+        <input style={{ display: 'none' }} id="files" accept=".pdf" type="file" name="resume" onChange={this.onChange} onBlur={() => {}} disabled={disabled} />
```
Comment (Jumer-Caragay) [inline src/components/InputTypes/FileInput.jsx]: Let's multiline these props

File: src/containers/Questionnaire.jsx [hunk 2]
```diff
@@ -151,4 +146,4 @@ Questionaire.propTypes = {
   }).isRequired,
 };
 
-export default connect(mapStateToProps, null)(requireAuth(Questionaire));
+export default connect(mapStateToProps, null)(requireAuth(Questionnaire));
```
Comment (Jumer-Caragay) [inline src/containers/Questionnaire.jsx]: Thanks for the renaming

File: src/components/InputTypes/MultiSelect.jsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ const MultiSelect = (props) => {
   function renderOptions() {
     return Object.keys(options).map((option) => {
```
Comment (Jumer-Caragay) [inline src/components/InputTypes/MultiSelect.jsx]: Honestly we should prob rename the option parameter to something better to differentiate between the key and the value

File: src/components/InputTypes/MultiSelect.jsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ const MultiSelect = (props) => {
...
       return (
-        <MenuItem key={option} value={option}>
+        <MenuItem key={option} value={options[option]}>
```
Comment (Jumer-Caragay) [inline src/components/InputTypes/MultiSelect.jsx]: I learned that to make react more concise about keys in map functions, you should do something like add the additional index parameter to the map ex Object.keys(options).map((option, i) and then make the key={`multi_select_${option}_${i}`}.
Although it's unlikely that there would be a key that would be equal to "Female" you want to make the key as close to what it is rendering alongside the additional i parameter in map

File: src/actions/index.js [hunk 5]
```diff
@@ -136,4 +134,28 @@ export const submitApp = (user, form) => (dispatch) => {
   //   .catch((error) => {
   //     dispatch({ type: types.SUBMISSION_FAIL, error });
   //   });
+  form.grad_date = form.grad_date.replace('-', '');
+  form.status = 0;
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: Funny part is status isn't being captured by our backend to indicate the status of the application, we need to address this in https://github.com/slohacks/attendee-application-back-end/blob/master/src/models/ApplicationModel.js

Comment (Jumer-Caragay) [reply src/actions/index.js]: Eventually you won't need to add the status since we will default it to 0 on submission

File: src/actions/index.js [hunk 6]
```diff
@@ -80,28 +80,26 @@ export function submitResponse(formProps) {
 
 export const uploadResume = (user, resume, onChange) => (dispatch) => {
   dispatch({ type: types.UPLOAD_RESUME_ATTEMPT });
-  // const storageRef = firebase
-  //   .storage()
-  //   .ref()
-  //   .child('resumes')
-  //   .child(`${user.uid}.pdf`);
+  const data = new FormData();
+  data.append('resume', resume);
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: https://gyazo.com/e4305905bd2f6c35106e5ee407e697a3, Check out this lint error

File: src/components/InputTypes/MultiSelect.jsx (hunk 4 above)
Comment (toyoun) [reply src/components/InputTypes/MultiSelect.jsx]: I tried this out, but I don't think it's necessary.

My reason is that when I tried to add a key in the `questionList.json` file with the same name as another, lint picked it up. Lint will catch any duplicate names we have in the keys within a single object, so uniqueness is practically guaranteed. Unless we want to also check for unique keys between different objects, which it won't pick up.

File: src/actions/index.js (hunk 5 above)
Comment (toyoun) [reply src/actions/index.js]: Kk. Lemme know when the model is updated and I'll remove it ASAP. :)

File: src/components/InputTypes/MultiSelect.jsx (hunk 3 above)
Comment (toyoun) [reply src/components/InputTypes/MultiSelect.jsx]: Renamed this to `choices`. Lemme know what you think!

File: src/actions/index.js [hunk 7]
```diff
@@ -136,4 +135,28 @@ export const submitApp = (user, form) => (dispatch) => {
   //   .catch((error) => {
   //     dispatch({ type: types.SUBMISSION_FAIL, error });
   //   });
+  formCopy.grad_date = form.grad_date.replace('-', '');
+  formCopy.status = 0;
```
Comment (Jumer-Caragay) [inline src/actions/index.js]: Remove this

File: src/components/InputTypes/MultiSelect.jsx (hunk 4 above)
Comment (toyoun) [reply src/components/InputTypes/MultiSelect.jsx]: Lint returns `Do not use Array index in keys` when attempting to add the index number as part of the key.

Screen Shot 2019-10-03 at 7 52 22 PM

So I will not be doing the map index change.

File: src/actions/index.js (hunk 5 above)
Comment (toyoun) [reply src/actions/index.js]: Removed the status variable.

**Pull request merged**